metricbeat/helper/sql/sql.go

```diff
@@ -0,0 +1,187 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/lib/pq"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/pkg/errors"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Db struct {
+	*sql.DB
+	logger *logp.Logger
+}
+
+type sqlRow interface {
+	Scan(dest ...interface{}) error
+	Next() bool
+	Columns() ([]string, error)
+	Err() error
+}
+
+// DB gets a client ready to query the database
+func DB(driver, uri string, l *logp.Logger) (*Db, error) {
```

Comment:
Not strong opinion but maybe `CreateDBClient` could be more handy for future code visitors.

Reply:
The reason I didn't call it `GetDb` or `CreateDb` is this https://golang.org/doc/effective_go.html#Getters

Reply:
:+1 for not having a `Get` prefix but how about `DBClient`?

Reply:
If the type is called `DbClient` at the package level, function cannot be called `DbClient` at the same package level :smile: So probably `NewDbClient` is the best candidate